framework/android/kodein-di-framework-android-x-viewmodel-savedstate/src/main/java/org/kodein/di/android/x/viewmodel/DIAware.kt

```diff
@@ -0,0 +1,86 @@
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.createViewModelLazy
+import androidx.lifecycle.*
+import org.kodein.di.DIAware
+import org.kodein.di.direct
+import org.kodein.type.generic
+
+
+/**
+ * Returns a [Lazy] property delegate to access the [AppCompatActivity]'s [ViewModel], which will be obtained from a Kodein DI
+ * factory with a [SavedStateHandle] argument.
+ *
+ * ```
+ * class MyComponentActivity : AppCompatActivity(), DIAware {
+ *     override val di: DI by closestDI()
+ *     private val viewmodel: MyViewModel by viewModelSavedStateFactory()
+ * }
+ * ```
+ * @param A Extension function applies to [AppCompatActivity] which implements [DIAware]
+ * @param VM The type of [ViewModel] to obtain
+ * @param tag The bound Kodein DI tag
+ * @throws IllegalArgumentException if accessed before the Activity is attached to the Application
+ * @throws DI.NotFoundException If no factory was found.
+ * @throws DI.DependencyLoopException When calling the factory, if the value construction triggered a dependency loop.
+ */
+@MainThread
+inline fun <A, reified VM> A.viewModelSavedStateFactory(
+        tag: Any? = null,
+): Lazy<VM> where A : AppCompatActivity, A : DIAware, VM : ViewModel {
+    val factoryProducer = { object : AbstractSavedStateViewModelFactory(this, null) {
+        override fun <T : ViewModel> create(key: String, modelClass: Class<T>, handle: SavedStateHandle): T {
+            val factory = direct.Factory(generic<SavedStateHandle>(), generic<VM>(), tag)
+            return factory(handle) as T
+        }
+    } }
+    return ViewModelLazy(VM::class, { viewModelStore }, factoryProducer)
+}
+
+/**
+ * Returns a [Lazy] property delegate to access a [ViewModel] by **default** scoped to this [Fragment], which will
+ * be obtained from a Kodein DI factory with a SavedStateHandle argument.
+ * ```
+ * class MyFragment : Fragment(), DIAware {
+ *     override val di: DI by closesDI()
+ *     private val viewmodel: MyViewModel by viewModelSavedStateFactory()
+ * }
+ * ```
+ *
+ * Default scope may be overridden with parameter [ownerProducer]:
+ * ```
+ * class MyFragment : Fragment(), DIAware {
+ *     override val di: DI by closestDI()
+ *     private val viewmodel: MyViewModel by viewModelSavedStateFactory ({requireParentFragment()})
+ * }
+ * ```
+ *
+ * @param F Extension function applies to [androidx.fragment.app.Fragment] which implements [org.kodein.di.DIAware]
+ * @param VM The type of [androidx.lifecycle.ViewModel] to obtain
+ * @param ownerProducer Optionally override the default ViewModel owner
+ * @param tag The bound Kodein DI tag
+ * @throws IllegalArgumentException if accessed before this Fragment is attached
+ * @throws DI.NotFoundException If no factory was found.
+ * @throws DI.DependencyLoopException When calling the factory, if the value construction triggered a dependency loop.
+ */
+@MainThread
+inline fun <F, reified VM> F.viewModelSavedStateFactory(
```

Comment:
Can you rename this as `viewModelSavedState` ? 
As the factory is an internal thing et it just returns a lazy viewModel.